packages/devtools/lib/src/timeline/timeline.dart

```diff
@@ -275,22 +295,29 @@ class TimelineScreen extends Screen {
     }
   }
 
-  /// Adds buttons to the timeline that will dump debug information to text
-  /// files and download them. This will only appear if either the
-  /// [debugTimeline] flag is true or the [debugCpuProfile] flag is true.
+  void _exportTimeline() {
+    // There will be a trailing comma on [debugTraceEvents]. Remove so that we
+    // do not have to deal with json formatting errors later.
+    final traceEvents = debugTraceEvents.isEmpty
+        ? ''
+        : debugTraceEvents.toString().replaceRange(
+            debugTraceEvents.length - 1, debugTraceEvents.length, '');
+    final String json = '{'
```

Comment:
I would create a large Map and then JSON encode rather than manually emitting JSON to avoid bugs.

Reply:
done.